docs/specifications/tests/Basic-TP/basic02.md

```diff
@@ -1,73 +1,219 @@
+follow the specification for DNS queries as specified in
+[DNS Query and Response Defaults]. The handling of the DNS responses on the DNS
+queries follow, unless otherwise specified below, what is specified for
+[DNS Response] in the same specification.
+
+
+1. Create a [DNS Query] with query type SOA and query name *Child Zone*
+   ("SOA Query").
+
+2. Create the following empty sets:
+   1. Name server name and IP ("Auth Response on SOA Query").
+   2. Name server name and IP ("Broken NS").
+   3. Name server name and IP ("NS not auth").
+   4. Name server name ("NS Cannot Resolve Into IP").
+   5. Name server name and IP ("No Respons From NS").
+   6. Name server name and IP plus [RCODE Name] ("Unexpecte RCODE").
+   7. Name server name and IP ("Delegation NS").
+
+3. Populate the set *Delegation NS* with name and IP address for the name
+   servers of the delegation of *Child Zone*.
+   1. If the test is an undelegated test, then:
+      1. Use *Undelegated NS*, *Undelegated Glue IP* and
+      *Undelegated Non-Glue IP*.
+      2. If any [out-of-bailiwick] name server name in the set has no IP address
+         then do a recursive lookup for address records (both IPv4 and IPv6) for
+         that name and add resolved addresses, if any, to the set.
+   2. Else, do:
+      1. Retrieve the NS records for *Child Zone* using [Method 2] and the IP
+         addresses ([glue records][glue record]) for any [in-bailiwick] name
+         servers using [Method 4].
+      2. Retrieve the IP addresses for any [out-of-bailiwick] name servers
+         using recursive lookup for address records (both IPv4 and IPv6) for
+         that name and add resolved addresses, if any, to the set.
+
+4. If the the *Delegation NS* set is empty, then do:
+   1. Output *[B03_NO_DELEGATION]* with *Child Zone* name.
+   2. Exit these test procedures.
+
+5. Else, for each name server name in the *Delegation NS* set do:
+   1. If the name server name has no IP address then do:
+      1. Add the name server name to the *NS Cannot Resolve Into IP*.
+      2. Go to next name server name.
+   2. If the name server name has multiple IP addresses, then repeat the loop
+      and the following steps for each IP address, else go to next name server
+      name i the loop.
+   3. Send *SOA Query* to the name server IP.
+   4. If there is no [DNS Response], then:
+      1. Add the name server name and IP address to the *No Response From NS*
+         set.
+      2. Go to next name server.
+   5. Else, if the AA flag is not set in the [DNS Response], then:
+      1. Add the name server name and IP address to the *NS not auth* set.
+      2. Go to next name server.
+   6. Else, if the [RCODE Name] is not "NoError" in the [DNS Response], then:
+      1. Add the name server name and IP address plus the [RCODE Name] to the
+         *Unexpecte RCODE* set.
+      2. Go to next name server IP address or next name server name.
```

Comment:
Go to next name server IP address or next name server name. -> Go to next name server IP address if available, else go to next name server name.

Reply:
Updated.